ifaddr/_posix.py

```diff
@@ -58,14 +58,56 @@ class ifaddrs(ctypes.Structure):
 else:
     IFF_MULTICAST = 1 << 12
 
+def get_interfaces_cmdline():
+    import os, re, ipaddress
+    ifaces = {}
+    curr_iface_name = None
+    curr_iface = None
+    try:
+        popen_pipe = os.popen('ifconfig')
+    except:
+        popen_pipe = []
+    for line in os.popen('ifconfig'):
+        match = re.match('^([a-zA-Z0-9]+):', line)
+        if match:
+            if curr_iface_name:
+                ifaces[curr_iface_name] = curr_iface
+            curr_iface_name = match.group(1)
+            curr_iface = {'multicast': False}
+            if 'MULTICAST' in line:
+                curr_iface['multicast'] = True
+            continue
+        match = re.match('\ *inet ([0-9\.]+)  netmask ([0-9\.]+)', line)
+        if match:
+            curr_iface[2] = []
+            curr_addr = {'addr': match.group(1), 'netmask': match.group(2)}
+            match = re.match('.* broadcast ([0-9\.]+)', line)
+            if match:
+                curr_addr['broadcast'] = match.group(1)
+            else:
+                match = re.match('.*  destination ([0-9\.]+)', line)
+                if match:
+                    curr_addr['peer'] = match.group(1)
+            curr_iface[2].append(curr_addr)
+            continue
+        match = re.match('\ *inet6 ([a-f0-9\.:]+)  prefixlen ([0-9]+)', line)
+        if match:
+            curr_iface[30] = []
+            network = ipaddress.IPv6Network(f'{match.group(1)}/{match.group(2)}', strict=False)
+            curr_addr = {'addr': match.group(1), 'prefix': match.group(2), 'netmask': f"{network.netmask.compressed}/{match.group(2)}"}
+            match = re.match('.* scopeid ([0-9x]+)', line)
+            if match:
+                curr_addr['scope'] = int(match.group(1), base=16)
+
+            curr_iface[30].append(curr_addr)
+    if curr_iface_name and curr_iface_name not in ifaces:
+        ifaces[curr_iface_name] = curr_iface
+    return ifaces
 
 def get_adapters(include_unconfigured: bool = False) -> Iterable[shared.Adapter]:
-    addr0 = addr = ctypes.POINTER(ifaddrs)()
-    retval = libc.getifaddrs(ctypes.byref(addr))
-    if retval != 0:
-        eno = ctypes.get_errno()
-        raise OSError(eno, os.strerror(eno))
-
+    #addr0 = addr = ctypes.POINTER(ifaddrs)()
+    #retval = libc.getifaddrs(ctypes.byref(addr))
+    retval = -1
```

Comment:
Could it be you forgot to uncomment your debug code here?

I'd expect this to rather be

```suggestion
    addr0 = addr = ctypes.POINTER(ifaddrs)()
    retval = libc.getifaddrs(ctypes.byref(addr))
```

so the fallback actually is a *fallback* if `libc.getifaddrs` fails and returns a non-zero `retval`? 🤔

Reply:
Yes, sorry